proxy/src/config.rs

```diff
@@ -380,3 +415,65 @@ fn parse<T, Parse>(strings: &Strings, name: &str, parse: Parse) -> Result<Option
         None => Ok(None),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn test_units<F: Fn(u64) -> Duration>(units: &[&str], to_duration: F) {
+        for v in &[1, 23, 456_789] {
+            let d = to_duration(*v);
+            for u in units {
+                let text = format!("{}{}", v, u);
+                assert_eq!(parse_duration(&text), Ok(d), "text=\"{}\"", text);
+
+                let text = format!("{} {}", v, u);
+                assert_eq!(parse_duration(&text), Ok(d), "text=\"{}\"", text);
+
+                let text = format!(" {}  {} ", v, u);
+                assert_eq!(parse_duration(&text), Ok(d), "text=\"{}\"", text);
+            }
+        }
+    }
+
+    #[test]
+    fn parse_duration_milliseconds() {
+        let units = ["ms", "msec", "msecs", "milli", "millis", "millisecond", "milliseconds"];
+        test_units(&units, |v| Duration::from_millis(v));
+    }
+
+    #[test]
+    fn parse_duration_seconds() {
+        let units = ["s", "sec", "secs", "second", "seconds"];
+        test_units(&units, |v| Duration::from_secs(v));
+    }
+
+    #[test]
+    fn parse_duration_minutes() {
+        let units = ["m", "min", "mins", "minute", "minutes"];
+        test_units(&units, |v| Duration::from_secs(v * 60));
+    }
+
+    #[test]
+    fn parse_duration_hours() {
+        let units = ["h", "hour", "hours"];
+        test_units(&units, |v| Duration::from_secs(v * 60 * 60));
+    }
+
+    #[test]
+    fn parse_duration_days() {
+        let units = ["d", "day", "days"];
+        test_units(&units, |v| Duration::from_secs(v * 60 * 60 * 24));
+    }
+
+    #[test]
+    fn parse_float_numbers_are_invalid() {
+        assert_eq!(parse_duration(".123 hours"), Err(ParseError::NotADuration));
+        assert_eq!(parse_duration("1.23 hours"), Err(ParseError::NotADuration));
+    }
+
+    #[test]
+    fn parse_invalid_unit() {
+        assert_eq!(parse_duration("12 moons"), Err(ParseError::NotADuration));
+    }
```

Comment:
We should also have a test for the input string that would cause a overflow, as well as for the maximum value that won't cause an overflow. Also negative numbers.